serializers/src/glTF/2.0/Extensions/KHR_materials_clearcoat.ts

```diff
@@ -0,0 +1,102 @@
+import { IMaterial } from "babylonjs-gltf2interface";
+import { IGLTFExporterExtensionV2 } from "../glTFExporterExtension";
+import { _Exporter } from "../glTFExporter";
+import { Material } from 'babylonjs/Materials/material';
+import { PBRMaterial } from 'babylonjs/Materials/PBR/pbrMaterial';
+import { PBRBaseMaterial } from 'babylonjs/Materials/PBR/pbrBaseMaterial';
+import { BaseTexture } from 'babylonjs/Materials/Textures/baseTexture';
+import { IKHRMaterialsClearcoat } from 'babylonjs-gltf2interface';
+import { PBRMetallicRoughnessMaterial } from "babylonjs/Materials/PBR/pbrMetallicRoughnessMaterial";
+
+const NAME = "KHR_materials_clearcoat";
+
+/**
+ * @hidden
+ */
+export class KHR_materials_clearcoat implements IGLTFExporterExtensionV2 {
+    /** Name of this extension */
+    public readonly name = NAME;
+
+    /** Defines whether this extension is enabled */
+    public enabled = true;
+
+    /** Defines whether this extension is required */
+    public required = false;
+
+    private _exporter: _Exporter;
+
+    private _wasUsed = false;
+
+    constructor(exporter: _Exporter) {
+        this._exporter = exporter;
+    }
+
+    public dispose() {
+    }
+
+    /** @hidden */
+    public get wasUsed() {
+        return this._wasUsed;
+    }
+
+    public postExportMaterialAdditionalTextures?(context: string, node: IMaterial, babylonMaterial: Material): BaseTexture[] {
+        let additionalTextures: BaseTexture[] = [];
+        if (babylonMaterial instanceof PBRMaterial) {
+            if (babylonMaterial.clearCoat.isEnabled) {
+                if (babylonMaterial.clearCoat.texture) {
+                    additionalTextures.push(babylonMaterial.clearCoat.texture);
+                }
+                if (!babylonMaterial.clearCoat.useRoughnessFromMainTexture && babylonMaterial.clearCoat.textureRoughness) {

```

Comment:
What about `remapF0OnInterfaceChange`?

Also, it looks like the textures will be ignored if the flags are not set like it is necessary for glTF. Either we should handle those cases or we should throw a warning or something to indicate that they weren't exported.

Reply:
> Also, it looks like the textures will be ignored if the flags are not set like it is necessary for glTF

Can you expand on this? I'm not sure what you mean here.


> What about remapF0OnInterfaceChange?

I dont understand, why would this affect the other serializable fields? It seems like something that we just can't handle.

Reply:
> > Also, it looks like the textures will be ignored if the flags are not set like it is necessary for glTF
>
> Can you expand on this? I'm not sure what you mean here.

When serializing, we shouldn't just drop textures if they are not compatible with glTF. Either we need to convert them to work with glTF or log/throw some kind of error/warning.

>> What about remapF0OnInterfaceChange?
> 
> I dont understand, why would this affect the other serializable fields? It seems like something that we can't express and should drop, unless I'm missing something.

The loader sets this flag. I don't know exactly what this flag does, but do we need to check it here? If this isn't set, is the material compatible with glTF?

Reply:
> When serializing, we shouldn't just drop textures if they are not compatible with glTF. Either we need to convert them to work with glTF or log/throw some kind of error/warning.

Can you be more specific? I still don't follow what you're getting at and why its relevant here

Reply:
If you're referring to `pbrClearCoatConfiguration._tintTexture` being dropped, I can add a warning indicating that it's not supported. This texture specifically has no analog.

Reply:
If `babylonMaterial.clearCoat.useRoughnessFromMainTexture` is true, then the roughness texture will be missing in the glTF. Right?

Reply:
I'm not familiar with this code to know whether this is expected or not. Looks like it's being used in `_finishMaterial` and `_exportTextureAsync` will be called on it. If `useRoughnessFromMainTexture` is true, it looks like the main `texture` won't be exported since it's not in the list. Even if you add it to the list, it will still be wrong since this won't match the glTF spec. Have you tried exporting a clearcoat material where `useRoughnessFromMainTexture` is true?

Reply:
Yes, it actually works when we point it to the proper texture. I don't believe we should be including this texture explicitly when useRoughessFromMainTexture is enabled, however. we should just try to find the texture if it exists. Similarly to the sheen implementation.

Reply:
Also added warnings for if we can't export features out of glTF's spec.